src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_FACTOR = 0.75f;
+    private int currentCapacity;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+        this.currentCapacity = table.length;
+        this.threshold = (int) (currentCapacity * DEFAULT_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getBucketIndex(key);
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(index, newNode, table);
```

Comment:
Private methods and classes should be after public ones in your class. The 'putValue' method should be placed after the public methods in the 'MyHashMap' class.